include/secp256k1_bulletproofs.h

```diff
@@ -0,0 +1,182 @@
+#ifndef _SECP256K1_BULLETPROOF_
+# define _SECP256K1_BULLETPROOF_
+
+# include "secp256k1.h"
+# include "secp256k1_generator.h"
+# include "secp256k1_rangeproof.h"
+
+# ifdef __cplusplus
+extern "C" {
+# endif
+
+/** Opaque structure representing a large number of NUMS generators */
+typedef struct secp256k1_bulletproof_generators secp256k1_bulletproof_generators;
+
+/** Version number used in header of circuit and circuit-assignment binary files */
+#define SECP256K1_BULLETPROOF_CIRCUIT_VERSION	1
+
+/* Maximum depth of 31 lets us validate an aggregate of 2^25 64-bit proofs */
+#define SECP256K1_BULLETPROOF_MAX_DEPTH 31
+
+/* Size of a hypothetical 31-depth rangeproof, in bytes */
+#define SECP256K1_BULLETPROOF_MAX_PROOF (160 + 36*32 + 7)
+
+/* Maximum memory, in bytes, that may be allocated to store a circuit representation */
+#define SECP256K1_BULLETPROOF_MAX_CIRCUIT (1024 * 1024 * 1024)
```

Comment:
include a hand-wavy reason for limit?

Reply:
This constant should just be dropped actually.